src/main/java/com/myorg/JavaCdk01BaselineStack.java

```diff
@@ -0,0 +1,46 @@
+package com.myorg;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import software.amazon.awscdk.services.iam.IManagedPolicy;
+import software.amazon.awscdk.services.iam.IPrincipal;
+import software.amazon.awscdk.services.iam.ManagedPolicy;
+import software.amazon.awscdk.services.iam.OpenIdConnectProvider;
+import software.amazon.awscdk.services.iam.Role;
+import software.amazon.awscdk.services.iam.WebIdentityPrincipal;
+import software.amazon.awscdk.Stack;
+import software.constructs.Construct;
+
+public class JavaCdk01BaselineStack extends Stack {
+  public JavaCdk01BaselineStack(final Construct scope, final String id) {
+    this(scope, id, null);
+  }
+
+  String issuerUrl = "https://token.actions.githubusercontent.com";
+  String sub = "repo:HenrikFricke/java-cdk-demo-01:ref:refs/heads/main";
+  List<String> audiences = Arrays.asList("sts.amazonaws.com");
+  List<IManagedPolicy> policies = Arrays.asList(ManagedPolicy.fromAwsManagedPolicyName("AdministratorAccess"));
+
+  @lombok.Builder
+  public JavaCdk01BaselineStack(final Construct scope, final String id, final JavaCdk01BaselineStackProps props) {
+    super(scope, id, props);
+
+    OpenIdConnectProvider githubProvider = OpenIdConnectProvider.Builder.create(this, "GitHubProvider")
+        .url(issuerUrl)
+        .clientIds(audiences)
+        .build();
+
+    Role.Builder.create(this, "GitHubActionsRole")
+        .description("Assumed by GitHub (aws-actions/configure-aws-credentials)")
+        .assumedBy(
+            new WebIdentityPrincipal(githubProvider.getOpenIdConnectProviderArn(),
+                Map.of(
+                    "StringEquals", Map.of(
```

Comment:
This is not working because I get `The method of(String, String, String, String) is undefined for the type Map`. Not sure why, need to investigate it next week.

Reply:
maybe this has to do with Map being generic and you'd have to specify somehow that this is a `Map<string, string>` becayse the compiler is not smart enough?

Reply:
Idk, it turns out it's just a problem on my machine and GitHub just compiles it.